aarch64-Linux

Waiting the ARM runner in public repositories https://github.com/github/roadmap/issues/970

When I tested, it failed only in arm runner.

https://github.com/kachick/nixpkgs-reviewing-workspace/actions/runs/12840033785/job/35808122405

```
  nix run 'github:NixOS/nixpkgs/nixpkgs-unstable#nixpkgs-review' -- pr '370188' --print-result --eval local
  shell: /usr/bin/bash -e {0}
```

```
unpacking 'github:NixOS/nixpkgs/ae06b9c2d[8](https://github.com/kachick/nixpkgs-reviewing-workspace/actions/runs/12840033785/job/35808122405#step:6:9)3cb5c8b12d7d0e32692e93d1379713' into the Git cache...
these 106 paths will be fetched (85.49 MiB download, 372.24 MiB unpacked):
  /nix/store/d1d5yay3jc2q83qqnii502c6wc4l6hg4-acl-2.3.2
  /nix/store/daiwlmh1pmifawc20x5a[9](https://github.com/kachick/nixpkgs-reviewing-workspace/actions/runs/12840033785/job/35808122405#step:6:10)gscjakzjymj-attr-2.5.2
  /nix/store/l9cfkqgkbsl03wwikf4s303s99pvs456-aws-c-auth-0.8.0
  /nix/store/nzkgyqv758cmmjg3lgh485f9wn9h33fh-aws-c-cal-0.8.0
  /nix/store/v4ipvjp85dnhhc9270rj6m783xn8x6sm-aws-c-common-0.[10](https://github.com/kachick/nixpkgs-reviewing-workspace/actions/runs/12840033785/job/35808122405#step:6:11).3
  /nix/store/2wkzsfj9dv138hap8x6cgw4hdrmdsx5q-aws-c-compression-0.3.0
  /nix/store/k5z9d5ki2g15i8ihsxgmps9xgivy20g6-aws-c-event-stream-0.5.0
  /nix/store/nd9hvv8fay1q9rl6na4j78jwfr69axh0-aws-c-http-0.9.2
  /nix/store/qd55slbns0ppks7d2ikhjngmrppv7lw1-aws-c-io-0.15.3
  /nix/store/axd2029y1vls9x3s9li6r8rz5f47c52r-aws-c-mqtt-0.[11](https://github.com/kachick/nixpkgs-reviewing-workspace/actions/runs/12840033785/job/35808122405#step:6:12).0
  /nix/store/nk4pdrys0fvbmvj36rxwinzkyzx3aqjw-aws-c-s3-0.7.1
  /nix/store/0frszwpdimpcv05yjimajvvv89s4czxl-aws-c-sdkutils-0.2.1
  /nix/store/s9jirra3qak1fvlry4x6qn586408yi6v-aws-checksums-0.2.2
  /nix/store/wdaagiliqh7ggmbif8v10x584bclwbsd-aws-crt-cpp-0.29.4
  /nix/store/qz0ajd3zlcnswlfb1w29vhry55p7sla2-aws-sdk-cpp-1.11.448
  /nix/store/rm5cm7wk2a6y17gz52ymh39d5fzai2lv-bash-5.2p37
  /nix/store/yp7rb7qsc3fnal6s3pwvifsylbrh0dxn-boehm-gc-8.2.8
  /nix/store/mfy9sihr0r7c75i8bdrb775rry6symc7-brotli-1.1.0-lib
  /nix/store/v9dgjpmg01f3sq377dj7i7wm25qf1l55-busybox-1.36.1
  /nix/store/n104rs7l21dx100763qffjdihipssj7q-bzip2-1.0.8
  /nix/store/lhd4sbpf3l9jxmb8sl6881l924q151j0-coreutils-9.5
  /nix/store/y340kr2q8x0y40krwzpv32x7d5c31jy7-curl-8.11.0
  /nix/store/fy0fxsgpirap81w5im4xi74aic2kx26b-editline-1.17.1
  /nix/store/6cljqf6dvax7llyrk9v5zlyfh1bsp94y-expat-2.6.4
  /nix/store/k7zrgs3l3ifa9qpz8i3ndaa0xvsmcb9n-gcc-14-20241116-lib
  /nix/store/g0fg0qk2[12](https://github.com/kachick/nixpkgs-reviewing-workspace/actions/runs/12840033785/job/35808122405#step:6:13)665jy2bmhrqvi7cqfzz90q-gcc-14-20241116-libgcc
  /nix/store/hyyij7f1i8daxbs6c4hy8rby70p01n1i-gdbm-1.24-lib
  /nix/store/i8i65pqq0m9r0xg8zh3hg0lkbaqv8zxg-gettext-0.22.5
  /nix/store/861v7354sw64ydk43jwdswwkqkf0dhq2-git-2.47.0
  /nix/store/4rj9xhq42v2b7nxf77xlkqbd20jzby0i-git-2.47.0-doc
  /nix/store/myf5swnbwk96kzs654k75s9fnfxhyaiz-glibc-2.40-36
  /nix/store/h3hcsmwwgsmnj11f7482lkrnycmlw3q8-gmp-with-cxx-6.3.0
  /nix/store/5cdhqqr2bq0dic7g7674rqnp287vvfdy-gnugrep-3.11
  /nix/store/6pa7svcbi274plssghalx3r74yfcifyb-gnused-4.9
  /nix/store/fl5mjwwd1ibphb011kmwqsl697n2abhr-gzip-1.[13](https://github.com/kachick/nixpkgs-reviewing-workspace/actions/runs/12840033785/job/35808122405#step:6:14)
  /nix/store/qf9i0kd239xzr52bsm20wd0sk4yqmxcq-keyutils-1.6.3-lib
  /nix/store/vrdf36r6vgzfv4jn3ydbhbybr1yfk6rg-krb5-1.21.3-lib
  /nix/store/vhb3388zkqcx3fdhabp7sghviwihmbzh-libarchive-3.7.7-lib
  /nix/store/rfbl02lcwr4iclv8ynw7lql72hda3d5s-libffi-3.4.6
  /nix/store/5v5mmrpm0g1g4ni6p3g17kxj9z4q1zhn-libgit2-1.8.4-lib
  /nix/store/9d0ffyk[14](https://github.com/kachick/nixpkgs-reviewing-workspace/actions/runs/12840033785/job/35808122405#step:6:15)5w5qh19y1nq3qy9lh2vsb0l-libidn2-2.3.7
  /nix/store/lkgh13r8drjch038xg01kqssqiqq1mqy-libpsl-0.21.5
  /nix/store/xlxijq0jdsvsaqf440k9n9c8b6m3ff99-libseccomp-2.5.5-lib
  /nix/store/c6k5bqcrnhxsnrpdnwjpw6pa9vl57il3-libsodium-1.0.20
  /nix/store/2z4kjrr9p58gdr014y1q1n00vnvr4fsm-libssh2-1.11.1
  /nix/store/6jfaxvj67kllh276c3x2sbsx1xbvr6gf-libunistring-1.3
  /nix/store/f75al8qx0gyiknjxy542c1grmga6kvw7-libxcrypt-4.4.36
  /nix/store/q8zi43mbxfainsx8n[15](https://github.com/kachick/nixpkgs-reviewing-workspace/actions/runs/12840033785/job/35808122405#step:6:16)sm70drh6npkkx-libxml2-2.13.5
  /nix/store/5drnkkcf9ca730rxyyyiwnbszl7gk1ay-lowdown-1.3.2-lib
  /nix/store/b4hp1wj6r7vlcniw1928dxq9hq09g1wq-mailcap-2.1.54
  /nix/store/8gpdml010vbh1jrvlhfvf6gvd9h1d1j2-mpdecimal-4.0.0
  /nix/store/njjb4k80lbk9c69biwa3vlsx4yi4rpkl-ncurses-6.4.20221231
  /nix/store/rj5q4v5pb105qv4sb819j5fa534i3hq2-nghttp2-1.64.0-lib
  /nix/store/kd22v13hg1nxwyxhx8ikgm5lx96l5mhs-nix-2.24.11
  /nix/store/x79sqpyy0404z5vg1k1r9cbazgf7gv81-nix-2.24.11-man
  /nix/store/fxcs0kp4b9k3l823z8rxppm6qgg7g54x-nixpkgs-review-3.0.1
  /nix/store/hr2y7kll8z06x6yqnhci7lz673addxiy-nss-cacert-3.107
  /nix/store/pj9m4jn66yd0jxpi93fl5cr6spk7lvjp-openssl-3.3.2
  /nix/store/lz523zndgb208f5qb7s0m5kl17x7k2bz-pcre2-10.44
  /nix/store/jy7xbl2k8679qwn4wx3xqrcxmwynq592-perl-5.40.0
  /nix/store/0v2z2zv6h96z6dv0awvbp9mbj9mp3nlv-perl5.40.0-Authen-SASL-2.1700
  /nix/store/ahqaxz8256s0m7g406k8hi9dyzj4drg9-perl5.40.0-CGI-4.59
  /nix/store/q2pq2xcaa3sm00h0zb305na5c7zhc67v-perl5.40.0-CGI-Fast-2.[16](https://github.com/kachick/nixpkgs-reviewing-workspace/actions/runs/12840033785/job/35808122405#step:6:17)
  /nix/store/rsw4f6q08lm45c9czc87nyvhnz8r9r43-perl5.40.0-Clone-0.46
  /nix/store/35wkm1lwymws563r98x9mp6al0ybqxih-perl5.40.0-Digest-HMAC-1.04
  /nix/store/1av5sswavmdgsdxyfsjf5wb3pwz95d7y-perl5.40.0-Encode-Locale-1.05
  /nix/store/1bg5nsgw12rdgznzbz8w5nnxv3q0nk8x-perl5.40.0-FCGI-0.82
  /nix/store/yi1h5w8vvjmq83dahfqhc57qhwbq32yf-perl5.40.0-FCGI-ProcManager-0.28
  /nix/store/xjzpdcszxdrwmii710sdph5kxfzy7akj-perl5.40.0-File-Listing-6.16
  /nix/store/izfbg57sn8ap5cfhi9prysa3xrzkx5xn-perl5.40.0-HTML-Parser-3.81
  /nix/store/pg42xhjsh7vvhikznixb6kzg3k0z9gy5-perl5.40.0-HTML-TagCloud-0.38
  /nix/store/3xj2jivddpii3c8l42yan8qgcbi3xjln-perl5.40.0-HTML-Tagset-3.20
  /nix/store/4x0c4vpxk9lm8wdxaavck62c1i980v7j-perl5.40.0-HTTP-CookieJar-0.014
  /nix/store/410z0agl727n4dcp3ssvhrpcy6swp3fi-perl5.40.0-HTTP-Cookies-6.10
  /nix/store/29dxbz6n69v31s6ghc13vfa358hnsmhc-perl5.40.0-HTTP-Daemon-6.16
  /nix/store/f35623v00p7xi1fygbz99kgg5r8c64xf-perl5.40.0-HTTP-Date-6.06
  /nix/store/khfzzapv8lvb7vhxhrh9n6xz9747bayh-perl5.40.0-HTTP-Message-6.45
  /nix/store/g9gjy4qz5aisxw1kpl87jdliayq1sycy-perl5.40.0-HTTP-Negotiate-6.01
  /nix/store/g5vwiv2yrjlyp5s4mk11r20zl7xq65ds-perl5.40.0-IO-HTML-1.004
  /nix/store/5f27i6vw22cb3qvlq5m1m85138wg8x7q-perl5.40.0-IO-Socket-SSL-2.083
  /nix/store/hh99m2s4022gqry2irwxlalbgj0zpg2m-perl5.40.0-LWP-MediaTypes-6.04
  /nix/store/zqdn87a591i889jgkk6faskmpab040fi-perl5.40.0-Mozilla-CA-20230821
  /nix/store/d4c2y1cnf5jxr9hw3ngsw6mxdsfbgd8s-perl5.40.0-Net-HTTP-6.23
  /nix/store/6xc3f2ym7lamkaq8x9i7fk5fys1bflr7-perl5.40.0-Net-SMTP-SSL-1.04
  /nix/store/kwv8wyjd0ilbcgg415pq9ygrs70rk797-perl5.40.0-Net-SSLeay-1.92
  /nix/store/y53354ccmls4vh6x2rm5gmy9yvq1dz3m-perl5.40.0-TermReadKey-2.38
  /nix/store/kq70djs9s91gkcb97fcs3hmxphsx69l1-perl5.40.0-Test-Fatal-0.0[17](https://github.com/kachick/nixpkgs-reviewing-workspace/actions/runs/12840033785/job/35808122405#step:6:18)
  /nix/store/jpkq2p3s6admlz1plz85ra1wj0v9qqp3-perl5.40.0-Test-Needs-0.002010
  /nix/store/vd6dy3bw2z3kdx85yny6ng7ajp96fiac-perl5.40.0-Test-RequiresInternet-0.05
  /nix/store/mfh9k7hkjipaq5mikik4fj7adp61as0w-perl5.40.0-TimeDate-2.33
Updating files:  56% (25606/45725)
Updating files:  56% (25814/45725)
Updating files:  57% (26064/45725)
Updating files:  58% (26521/45725)
Updating files:  59% (26978/45725)
Updating files:  60% (27435/45725)
Updating files:  61% (27893/45725)
Updating files:  62% (28350/45725)
Updating files:  63% (28807/45725)
Updating files:  64% (29264/45725)
Updating files:  65% (29722/45725)
Updating files:  66% (30179/45725)
Updating files:  67% (30636/45725)
Updating files:  67% (31065/45725)
Updating files:  68% (31093/45725)
Updating files:  69% (31551/45725)
Updating files:  70% (32008/45725)
Updating files:  71% (32465/45725)
Updating files:  72% (32922/45725)
Updating files:  73% (33380/45725)
Updating files:  74% (33837/45725)
Updating files:  75% (34294/45725)
Updating files:  76% (34751/45725)
Updating files:  77% (35209/45725)
Updating files:  77% (35330/45725)
Updating files:  78% (35666/45725)
Updating files:  79% (36123/45725)
Updating files:  80% (36580/45725)
Updating files:  81% (37038/45725)
Updating files:  82% (37495/45725)
Updating files:  83% (37952/45725)
Updating files:  84% (38409/45725)
Updating files:  85% (38867/45725)
Updating files:  86% (39324/45725)
Updating files:  87% (39781/45725)
Updating files:  88% (40238/45725)
Updating files:  88% (40397/45725)
Updating files:  89% (40696/45725)
Updating files:  90% (41153/45725)
Updating files:  91% (41610/45725)
Updating files:  92% (42067/45725)
Updating files:  93% (42525/45725)
Updating files:  94% (42982/45725)
Updating files:  95% (43439/45725)
Updating files:  96% (43896/45725)
Updating files:  97% (44354/45725)
Updating files:  98% (44811/45725)
Updating files:  98% (45144/45725)
Updating files:  99% (45268/45725)
Updating files: 100% (45725/45725)
Updating files: 100% (45725/45725), done.
HEAD is now at a[18](https://github.com/kachick/nixpkgs-reviewing-workspace/actions/runs/12840033785/job/35808122405#step:6:19)5fbd6f perlPackages.(various): fix strictDeps / cross build (#371816)
corrupted size vs. prev_size
https://github.com/NixOS/nixpkgs/pull/370188 failed to build: Failed to list packages: nix-env failed with exit code -6
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs --depth=1 master:refs/nixpkgs-review/0 pull/370188/merge:refs/nixpkgs-review/1
$ git worktree add /home/runner/.cache/nixpkgs-review/pr-370188/nixpkgs a185fbd6fef185e5f06f0e1d9b1dc4fbda4df46e
Local evaluation for computing rebuilds
$ nix-env --extra-experimental-features no-url-literals --option system aarch64-linux -f <nixpkgs> --nix-path nixpkgs=/home/runner/.cache/nixpkgs-review/pr-370188/nixpkgs nixpkgs-overlays=/tmp/tmph5bdqum6 -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
$ git worktree remove -f /home/runner/.cache/nixpkgs-review/pr-370188/nixpkgs
Error: Process completed with exit code 1.
```

--- a/.github/workflows/nixpkgs-review.yml
+++ b/.github/workflows/nixpkgs-review.yml
@@ -18,11 +18,11 @@ on:
         #   - https://github.com/actions/toolkit/issues/184
         type: string
         # - ubuntu-24.04 # x86_64
-        # - ubuntu-24.04-arm # aarch64-Linux # It raising errors. See GH-43
+        # - ubuntu-24.04-arm # aarch64-Linux
         # - macos-15 # aarch64
         # - macos-13 # x86_64
         default: |
-          [ "ubuntu-24.04", "macos-15", "macos-13" ]
+          [ "ubuntu-24.04", "ubuntu-24.04-arm", "macos-15", "macos-13" ]
 
 # env:
 #   pr-number: '42' # Enable only when developing

--- a/.github/workflows/nixpkgs-review.yml
+++ b/.github/workflows/nixpkgs-review.yml
@@ -1,9 +1,9 @@
 name: 👁️
 # https://stackoverflow.com/questions/71155641/github-actions-how-to-view-inputs-for-workflow-dispatch
-run-name: 'NixOS/nixpkgs#${{ inputs.pr-number }}'
-# run-name: '[Experimental]' # Enable only when developing
+# run-name: 'NixOS/nixpkgs#${{ env.pr-number }}'
+run-name: '[Experimental]' # Enable only when developing
 on:
-  # pull_request: # Enable only when developing
+  pull_request: # Enable only when developing
   workflow_dispatch:
     inputs:
       pr-number:
@@ -24,15 +24,15 @@ on:
         default: |
           [ "ubuntu-24.04", "ubuntu-24.04-arm", "macos-15", "macos-13" ]
 
-# env:
-#   pr-number: '42' # Enable only when developing
+env:
+  pr-number: '395562' # Enable only when developing
 
 jobs:
   review:
     strategy:
       fail-fast: false
       matrix:
-        runner: '${{ fromJson(inputs.runners) }}'
+        runner: ${{ fromJson('[ "ubuntu-24.04", "ubuntu-24.04-arm", "macos-15" ]') }}
     runs-on: ${{ matrix.runner }}
     # timeout-minutes: 180 # Any limitations might not fit for this repository. So temporary disabling
     steps:
@@ -64,12 +64,12 @@ jobs:
           GITHUB_TOKEN: '${{ github.token }}'
         run: | # shell
           #
-          nix run 'github:NixOS/nixpkgs/nixpkgs-unstable#nixpkgs-review' -- pr '${{ inputs.pr-number }}' --print-result
+          nix run 'github:NixOS/nixpkgs/nixpkgs-unstable#nixpkgs-review' -- pr '${{ env.pr-number }}' --print-result
 
       - name: Upload artifact
         uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
         with:
-          name: 'nixpkgs-review-files-pr-${{ inputs.pr-number }}-${{ runner.arch }}-${{ runner.os }}'
+          name: 'nixpkgs-review-files-pr-${{ env.pr-number }}-${{ runner.arch }}-${{ runner.os }}'
           path: '${{ steps.get-meta.outputs.cache_dir }}/nixpkgs-review/'
           if-no-files-found: 'error'
 

--- a/.github/workflows/nixpkgs-review.yml
+++ b/.github/workflows/nixpkgs-review.yml
@@ -1,9 +1,9 @@
 name: 👁️
 # https://stackoverflow.com/questions/71155641/github-actions-how-to-view-inputs-for-workflow-dispatch
-# run-name: 'NixOS/nixpkgs#${{ env.pr-number }}'
-run-name: '[Experimental]' # Enable only when developing
+run-name: 'NixOS/nixpkgs#${{ inputs.pr-number }}'
+# run-name: '[Experimental]' # Enable only when developing
 on:
-  pull_request: # Enable only when developing
+  # pull_request: # Enable only when developing
   workflow_dispatch:
     inputs:
       pr-number:
@@ -24,15 +24,15 @@ on:
         default: |
           [ "ubuntu-24.04", "ubuntu-24.04-arm", "macos-15", "macos-13" ]
 
-env:
-  pr-number: '395562' # Enable only when developing
+# env:
+#   pr-number: '42' # Enable only when developing
 
 jobs:
   review:
     strategy:
       fail-fast: false
       matrix:
-        runner: ${{ fromJson('[ "ubuntu-24.04", "ubuntu-24.04-arm", "macos-15" ]') }}
+        runner: '${{ fromJson(inputs.runners) }}'
     runs-on: ${{ matrix.runner }}
     # timeout-minutes: 180 # Any limitations might not fit for this repository. So temporary disabling
     steps:
@@ -64,12 +64,12 @@ jobs:
           GITHUB_TOKEN: '${{ github.token }}'
         run: | # shell
           #
-          nix run 'github:NixOS/nixpkgs/nixpkgs-unstable#nixpkgs-review' -- pr '${{ env.pr-number }}' --print-result
+          nix run 'github:NixOS/nixpkgs/nixpkgs-unstable#nixpkgs-review' -- pr '${{ inputs.pr-number }}' --print-result
 
       - name: Upload artifact
         uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
         with:
-          name: 'nixpkgs-review-files-pr-${{ env.pr-number }}-${{ runner.arch }}-${{ runner.os }}'
+          name: 'nixpkgs-review-files-pr-${{ inputs.pr-number }}-${{ runner.arch }}-${{ runner.os }}'
           path: '${{ steps.get-meta.outputs.cache_dir }}/nixpkgs-review/'
           if-no-files-found: 'error'
 

--- a/README.md
+++ b/README.md
@@ -22,12 +22,12 @@ As I understand that, Which github provided runner matches the Nix supported [sy
 
 Excluding _large_ runners here. Which cannot be used in free plan.
 
-| Nix                                                                               | GitHub           |
-| --------------------------------------------------------------------------------- | ---------------- |
-| x86_64-linux                                                                      | ubuntu-24.04     |
-| [aarch64-Linux](https://github.com/kachick/nixpkgs-reviewing-workspace/issues/43) | ubuntu-24.04-arm |
-| x86_64-darwin                                                                     | macos-13         |
-| aarch64-darwin                                                                    | macos-15         |
+| Nix            | GitHub                                                                                                       |
+| -------------- | ------------------------------------------------------------------------------------------------------------ |
+| x86_64-linux   | ubuntu-24.04                                                                                                 |
+| aarch64-Linux  | [ubuntu-24.04-arm](https://github.com/actions/partner-runner-images/blob/main/images/arm-ubuntu-24-image.md) |
+| x86_64-darwin  | macos-13                                                                                                     |
+| aarch64-darwin | macos-15                                                                                                     |
 
 ## Usage
 

--- a/sort.rb
+++ b/sort.rb
@@ -9,7 +9,7 @@ module Sorter
   # List of tiers: https://github.com/NixOS/nixpkgs/blob/nixos-24.11/lib/systems/flake-systems.nix
   NIX_BY_ASSET_NAME = {
     Runner.new(arch: 'X64', os: 'Linux').asset_name => Nix.new(platform: 'x86_64-linux', tier: 1),
-    # Runner.new(arch: 'ARM64', os: 'Linux').asset_name => Nix.new(platform: 'aarch64-linux', tier: 2), # TODO: Adjust after GH-43
+    Runner.new(arch: 'ARM64', os: 'Linux').asset_name => Nix.new(platform: 'aarch64-linux', tier: 2),
     Runner.new(arch: 'X64', os: 'macOS').asset_name => Nix.new(platform: 'x86_64-darwin', tier: 2),
     Runner.new(arch: 'ARM64', os: 'macOS').asset_name => Nix.new(platform: 'aarch64-darwin', tier: 3)
   }
@@ -18,8 +18,16 @@ module Sorter
     path.split('/').detect { |tree| /\Anixpkgs-review-files-.+\z/.match?(tree) }.slice(Regexp.union(NIX_BY_ASSET_NAME.keys))
   end
 
+  def self.parse_asset_name(asset_name)
+    arch, os = *asset_name.split('-')
+    { arch:, os: }
+  end
+
   def self.priority(path)
-    NIX_BY_ASSET_NAME.fetch(extract_asset_name(path)).tier
+    asset_name = extract_asset_name(path)
+    nix_tier = NIX_BY_ASSET_NAME.fetch(asset_name).tier
+    my_favor = parse_asset_name(asset_name).fetch(:os) == 'Linux' ? 0 : Float::INFINITY
+    [nix_tier, my_favor]
   end
 end
 

